pkg/util/net/grpc/BUILD.bazel

```diff
@@ -9,5 +9,7 @@ go_library(
         "//pkg/util:go_default_library",
         "//staging/src/kubevirt.io/client-go/log:go_default_library",
         "@org_golang_google_grpc//:go_default_library",
```

Comment:
Just realized I was overcomplicating this by trying to look at the binary.
Here we explicitly ask for `"@org_golang_google_grpc//:go_default_library",`
which means we want the WORKSPACE entry that was pinned to 1.30 prior to this PR

Which is an issue on it's own since we were making PRs like https://github.com/kubevirt/kubevirt/pull/10673, disregarding the WORKSPACE entry.
I think this justifies a release note saying that the `grpc from go.mod is now correctly shipped in release images`

Reply:
I think it might be a better idea to pivot this PR to address the issue in its entirety in that case, as I have yet to check whether grpc is the only external dependency affected by this, WDYT?

Reply:
```bash
$ grep -Rw go_repository WORKSPACE -A1 | grep 'name\s*=' | grep -Ev "winrm" | cut -d'"' -f2 | xargs -I {} sh -c "git grep {} | wc -l"
1
1
1
1
1
1
1
1
1
25
1
1
1
1
```

Seems like only grpc-go is used regardless of whether there are other gazelle dependencies.

Reply:
FYI I believe this issue sums it up perfectly: https://github.com/bazelbuild/bazel-gazelle/issues/1115.

Reply:
> I think it might be a better idea to pivot this PR to address the issue in its entirety in that case, as I have yet to check whether grpc is the only external dependency affected by this, WDYT?

Yeah I think that makes sense



> Seems like only grpc-go is used regardless of whether there are other gazelle dependencies.

I think we should try delete grpc-go and other unused go_repositories from the WORKSPACE file together with the scattered `-exclude vendor/google.golang.org/grpc`, hopefully with @rmohr's blessing.
(If you don't want to do this in this PR, that's fine)

> FYI I believe this issue sums it up perfectly: [bazelbuild/bazel-gazelle#1115](https://github.com/bazelbuild/bazel-gazelle/issues/1115).

Yeah I see, I was hyperfocused on there being two places where we indicate the grpc dep version so neglected the gazelle ordering issue

Reply:
> I think we should try delete grpc-go and other unused go_repositories from the WORKSPACE file together with the scattered -exclude vendor/google.golang.org/grpc, hopefully with @rmohr's blessing.
(If you don't want to do this in this PR, that's fine)

I'd prefer to leave it for another PR to avoid a Chesterton's Fence scenario. 
I'm pretty sure that using vendored grpc (and also outright removing all of the external dependencies) may cause some issues that extend beyond the scope of this PR (e.g. aligning `pkg/hooks`).

Reply:
sure
/lgtm
/cc @mhenriks
attempt at TL DR https://github.com/kubevirt/kubevirt/pull/12617#discussion_r1722302634

Reply:
> I'm pretty sure that using vendored grpc (and also outright removing all of the external dependencies) may cause some issues that extend beyond the scope of this PR (e.g. aligning pkg/hooks).

Yeah, I am actually not 100% sure this is easy to solve without going to Bazel 7 (with bazel module support). So just using the correct version initially sounds good to me, and go from there.

Reply:
So how come we have to bring those deps as part of WORKSPACE to begin with?